### PR TITLE
Add deepl-api-usage extension

### PR DESCRIPTION
**Pull request by DophinL**
## Description

I have multiple DeepL keys corresponding to different DeepL accounts. They are used when using the Chrome extension, but sometimes it stops working. I suspect that I have run out of usage limit, so I need to check the usage on the DeepL official website. Sometimes, I even need to switch accounts, which is quite troublesome. Therefore, I hope to use this plugin to manage the usage of my multiple accounts and quickly select one with remaining quota for translation purposes.

## Screencast

deepl-api-usage-1
deepl-api-usage-2
deepl-api-usage-3
deepl-api-usage-4

## Checklist

- [x] I read the [extension guidelines](https://developers.raycast.com/basics/prepare-an-extension-for-store)
- [x] I read the [documentation about publishing](https://developers.raycast.com/basics/publish-an-extension)
- [x] I ran `npm run build` and [tested this distribution build in Raycast](https://developers.raycast.com/basics/prepare-an-extension-for-store#metadata-and-configuration)
- [x] I checked that files in the `assets` folder are used by the extension itself
- [x] I checked that assets used by the `README` are placed outside of the `metadata` folder

### CODE REVIEW

File: extensions/deepl-api-usage/src/index.tsx [hunk 1]
```diff
@@ -0,0 +1,293 @@
...
+const colorMapper = cond([
+  [(percentage: number) => percentage < 0.1, () => Color.SecondaryText],
+  [(percentage: number) => percentage < 0.4, () => Color.Green],
+  [(percentage: number) => percentage < 0.6, () => Color.Blue],
+  [(percentage: number) => percentage < 0.8, () => Color.Orange],
+  [stubTrue, () => Color.Red],
+]);
+
+export default function RecordList() {
+  const [records, setRecords] = useState<Record[]>([]);
```
[the remark this answers is not in the document]
Comment (DophinL) [reply extensions/deepl-api-usage/src/index.tsx:41]: thx, good suggestion!

File: extensions/deepl-api-usage/src/index.tsx [hunk 2]
```diff
@@ -0,0 +1,293 @@
...
+                          const updatedRecords = records.map((r) => (r.id === record!.id ? record : r));
+                          await LocalStorage.setItem("records", JSON.stringify(updatedRecords));
+                          showToast(Toast.Style.Success, "Record Modified");
+                          loadRecords();
+                        }}
+                      />
+                    }
+                  />
+                  {addRecordNode}
+                  <Action icon={Icon.Trash} title="Delete Record" onAction={() => handleDelete(record.id)} />
```
[the remark this answers is not in the document]
Comment (DophinL) [reply extensions/deepl-api-usage/src/index.tsx:158]: thx, good suggestion!

File: extensions/deepl-api-usage/src/index.tsx (hunk 1 above)
[the remark this answers is not in the document]
Comment (DophinL) [reply extensions/deepl-api-usage/src/index.tsx:42]: I've read that I use localStorage itself, and will setState as soon as the localStorage content is fetched, which is still very fast from practical experience. If I use useCachedState again here, will it be a bit redundant?

**Comment by pernielsentikaer** [comment]
Hi 👋

I added `useForm` but didn't have an account to test with, do you mind testing it or provide me a test apikey I can test with 🙂

**Comment by DophinL** [comment]
> Hi 👋
>
> I added `useForm` but didn't have an account to test with, do you mind testing it or provide me a test apikey I can test with 🙂

Thanks for helping me optimize, learned some more best practices haha! Let me test it out.

**Comment by DophinL** [comment]
At first when I created the plugin, I did it via Create Extention inside the raycast command.
It creates a repository and then I git init myself.
But today, in order to pull your code, I cloned the fork's repository, made changes to the ext/deepl-api-usage branch in it, and finally `npm run publish`. But it reports an error, not sure if this is normal?

image
@pernielsentikaer

**Comment by pernielsentikaer** [comment]
Did you try to run the command it tell you to run in the screenshot? 

Seems like you pushed everything now, right?

**Comment by DophinL** [comment]
> Did you try to run the command it tell you to run in the screenshot?
>
> Seems like you pushed everything now, right?

Yes, I believe it’s all good now and there won’t be any new submissions. @pernielsentikaer

**Comment by DophinL** [comment]
I am writing the second plugin, which involves summarizing the content of the RSS Feed through LLM. I would like to also support Raycast AI as a provider. Can you provide a test API key? @pernielsentikaer

**Pull request merged**